consensus/hotstuff/safety_rules.go

```diff
@@ -0,0 +1,30 @@
+package hotstuff
+
+import (
+	"github.com/onflow/flow-go/consensus/hotstuff/model"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// SafetyRules produces votes for the given block according to voting rules.
+type SafetyRules interface {
+	// ProduceVote takes a block proposal and current view, and decides whether to vote for the block.
+	// Returns:
+	//  * (vote, nil): On the _first_ block for the current view that is safe to vote for.
+	//    Subsequently, voter does _not_ vote for any other block with the same (or lower) view.
+	//  * (nil, model.NoVoteError): If the safety module decides that it is not safe to vote for the given block.
+	//    This is a sentinel error and _expected_ during normal operation.
+	// All other errors are unexpected and potential symptoms of uncovered edge cases or corrupted internal state (fatal).
+	ProduceVote(proposal *model.Proposal, curView uint64) (*model.Vote, error)
+	// ProduceTimeout takes current view, highest locally known QC and TC and decides whether to produce timeout for current view.
+	// Returns:
+	//  * (timeout, nil): On the _first_ block for the current view that is safe to vote for.
+	//    Subsequently, voter does _not_ vote for any other block with the same (or lower) view.
```

Comment:
same logic as with voting, to avoid timeout equivocation my plan is to cache latest created timeout and return it for any subsequent call. If for instance by mistake we pass a lower view that we have already timed out == it's an error.